rules/0_preprocessing/scripts/make_input_lists.py

```diff
@@ -49,11 +62,57 @@ def get_read_naming_patterns(directory):
 paths  = {}
 layout = {}
 
-
 ## Check for the presence of a metadata table in in config, either for local reads ("local_samples") or distant reads ("sra_samples")
 if "local_samples" not in config.keys() and "sra_samples" not in config.keys():
     raise ValueError("No samples defined in the config file")
 
+
+##[Modification 2] Check the sample tsv for the following vaiables in config: 'sample_label', 'grouping_column', 'run_column'.
+##Sometimes there could be a mismatch between input column name for 'sample_label', 'grouping_column', 'run_column' from config file 
+##and sample metadata TSV file. This error is being handeled in the following code:
+
+# read sample dataframe from config file
+df = pandas.read_csv(config["local_samples"], sep="\t", index_col=0)
+
+# get metadata from sample dataframe
+df_colnames = df.columns
+
+
+to_check = ['sample_label', 'grouping_column', 'run_column'] # a list of important columns that should be in the dataframe
+x = (dict((k, config[k]) for k in to_check))
+
+check_list = x.values() # list containing the user defined columns from the metadata
+
+# columns to check in the sample metadata
+df_check_columns = []
+for i in df_colnames:
```

Comment:
Which difference with "check_list" above?

Reply:
check_list  contains the three variables from config file and df_check_columns contains the name assigned by the user to the columns! (actually, we are trying to compare the values assigned to config variables and comparing it with the actual column names in the sample metadata!

To do this:
1) I have created 2 lists and take a difference between the two (config variables has a mistake in assigning the column names or whatever)
2) then take the config key for the variable that differs between config file and sample metadata